### ISSUE TITLE
Cannot build with versions >1.13.4

### ISSUE THREAD

**Issue by angelo-v** [inrupt/solid-client-authn-js#3052]
### Search terms you've used

polyfill, esbuild, undici

### Impacted package

Which packages do you think might be impacted by the bug ?

- [x] solid-client-authn-browser
- [ ] solid-client-authn-node
- [ ] solid-client-authn-core
- [ ] oidc-client-ext
- [ ] Other (please specify): ...

### Bug description

Since version 1.14 @inrupt/solid-client-authn-browser seems to depend on a lot of node internals. I am getting several errors when trying to build my app with esbuild, like these:

```
✘ [ERROR] Could not resolve "perf_hooks"

    ../node_modules/undici/lib/fetch/util.js:5:32:
      5 │ const { performance } = require('perf_hooks')
        ╵                                 ~~~~~~~~~~~~

  The package "perf_hooks" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "platform: 'node'" to do that, which will remove this error.

✘ [ERROR] Could not resolve "assert"

    ../node_modules/undici/lib/fetch/body.js:19:23:
      19 │ const assert = require('assert')
         ╵                        ~~~~~~~~

  The package "assert" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "platform: 'node'" to do that, which will remove this error.

✘ [ERROR] Could not resolve "util/types"

    ../node_modules/undici/lib/fetch/body.js:21:48:
      21 │ const { isUint8Array, isArrayBuffer } = require('util/types')

```

It worked fine with v1.13.4. Newer version seem to have introduced universal-fetch, which depends on undici, that causes the errors:

```
▶ npm ls undici   
  +-- @inrupt/solid-client-authn-browser@1.16.0
  | `-- @inrupt/universal-fetch@1.0.2
  |   `-- undici@5.22.1
```

The README only mentions to install the buffer polyfill, which I did. If other polyfills are required with the new version this should be documented.

### To Reproduce

1. Install esbuild and @inrupt/solid-client-authn-browser@1.16.0
2. Create file test.js:

```
import { login } from "@inrupt/solid-client-authn-browser";

function test() {
  login();
}

```

3. try to bundle it using esbuild: `esbuild --bundle test.js`

### Expected result

Successful build.

### Actual result

Errors shown above.

### Environment

Please run

npx envinfo --system --npmPackages --binaries --npmGlobalPackages --browsers

in your project folder and paste the output here:

```
$ npx envinfo --system --npmPackages --binaries --npmGlobalPackages --browsers
  System:
    OS: Linux 6.2 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 4.02 GB / 31.05 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 19.6.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.4.0 - /usr/local/bin/npm
  npmPackages:
    @inrupt/solid-client-authn-browser: 1.16.0 => 1.16.0 
    buffer: ^6.0.3 => 6.0.3 
    esbuild: ^0.18.11 => 0.18.11 
  npmGlobalPackages:
    corepack: 0.15.3
    npm: 9.4.0

```

**Comment by jeswr** [comment]
This should be fixed by the combination of the following 2 pieces of work:
 - https://github.com/inrupt/universal-fetch/pull/268
 - https://github.com/inrupt/solid-client-authn-js/pull/3049
 
So this should be solved in the next release.

If you want to use this in the meantime you should be able to install the package by doing the following

```
gh repo clone inrupt/solid-client-authn-js
cd ./solid-client-authn-js
npm i
npm run build
```

and then installing `@inrupt/solid-client-authn-browser` by npm installing the relative path from your package to the `packages/browser` folder.

**Comment by angelo-v** [comment]
Unfortunately I am still getting the errors with v1.17.0

```
> esbuild --bundle test.js

✘ [ERROR] Could not resolve "net"

    node_modules/undici/lib/core/connect.js:3:20:
      3 │ const net = require('net')
        ╵                     ~~~~~

  The package "net" wasn't found on the file system but is built into node. Are you trying to bundle
  for node? You can use "--platform=node" to do that, which will remove this error.

✘ [ERROR] Could not resolve "assert"

    node_modules/undici/lib/core/util.js:3:23:
      3 │ const assert = require('assert')
        ╵                        ~~~~~~~~

  The package "assert" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "--platform=node" to do that, which will remove this error.

✘ [ERROR] Could not resolve "util"

    node_modules/undici/lib/mock/mock-client.js:3:30:
      3 │ const { promisify } = require('util')
        ╵                               ~~~~~~

  The package "util" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "--platform=node" to do that, which will remove this error.

✘ [ERROR] Could not resolve "assert"

    node_modules/undici/lib/fetch/request.js:31:23:
      31 │ const assert = require('assert')
         ╵                        ~~~~~~~~

  The package "assert" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "--platform=node" to do that, which will remove this error.

✘ [ERROR] Could not resolve "stream"

    node_modules/undici/lib/api/api-pipeline.js:7:12:
      7 │ } = require('stream')
        ╵             ~~~~~~~~

  The package "stream" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "--platform=node" to do that, which will remove this error.

✘ [ERROR] Could not resolve "async_hooks"

    node_modules/undici/lib/api/api-pipeline.js:14:34:
      14 │ const { AsyncResource } = require('async_hooks')
         ╵                                   ~~~~~~~~~~~~~

  The package "async_hooks" wasn't found on the file system but is built into node. Are you trying
  to bundle for node? You can use "--platform=node" to do that, which will remove this error.

6 of 70 errors shown (disable the message limit with --log-limit=0)
```

```
>  npm ls undici
`-- @inrupt/solid-client-authn-browser@1.17.0
  `-- @inrupt/universal-fetch@1.0.2
    `-- undici@5.22.1
```

**Comment by jeswr** [comment]
If you are building for browser then `universal-fetch` should not be trying to pull in undici and instead just using the globally available fetch [here](https://github.com/inrupt/universal-fetch/blob/main/src/index-browser.ts) which is pointed to by the `browser` entry in the bundler.

Perhaps the use of browser-entry imports in your bundler entry needs to be opted into like we do [here](https://github.com/inrupt/solid-client-authn-js/blob/0f5275fe71f0d4d868d9918f231e7af49640e509/packages/browser/rollup.config.mjs#L34-L35) with rollup?

**Comment by angelo-v** [comment]
I think esbuild chooses the correct entrypoint. After some experimention I come to the conclusion that the problem comes from the `exports` field from `@inrupt/universion-fetch`, which does not differ between browser and node. Monkey patching it to the following allows me to build fine:

```
  "exports": {
    ".": {
      "browser": {
        "require": "./dist/index-browser.js",
        "import": "./dist/index-browser.mjs"
      },
      "default": {
        "require": "./dist/index.js",
        "import": "./dist/index.mjs"
      }
    }
  },
```

On the other hand I do not see why those exports are needed at all, since the package is not self-referencing. You could check if they can be removed alltogether. Removing the `exports` field also leads to a working build on my side.

**Comment by jeswr** [comment]
@angelo-v this is fixed as of universal-fetch v1.0.3

**Comment by angelo-v** [comment]
Seems to work, thanks!
